### ISSUE TITLE
Search for genes by symbol API returns plain text, not JSON

### ISSUE THREAD

**Issue by hlapp** [phenoscape/phenoscape-kb-services#29]
The [Search for genes by symbol API](http://docs.phenoscapekb.apiary.io/#reference/genes/gene-info) returns plain text, not JSON as the documentation advertises:

```
 IRI    label   taxon
http://zfin.org/ZDB-GENE-040426-731 brpf1   zebrafish
http://www.informatics.jax.org/marker/MGI:1926033   Brpf1   mouse     
```

It really should return the advertised JSON.

**Comment by balhoff** [comment]
You need to ask for JSON. Can you add an 'Accept' header with value 'application/json'?

**Comment by hlapp** [comment]
Jim - this is using Apiary. Any reason this method would behave differently than the others, which do return JSON?

**Comment by balhoff** [comment]
Most methods do not support multiple formats, so you get one result even if you don't send an `Accept` header. If you want to test different formats, go to the "Console" tab in Apiary, click "Headers", and add the header there before sending the request. In the documentation for this method, you can see that there are two MIME types available as Response examples.

**Comment by hlapp** [comment]
OK - this works. However, why is plain text the default (rather than requiring text/plain as the Accept header) and not JSON? Shouldn't that be JSON? Do you see how that's confusing and results in inconsistent behavior?

**Comment by hlapp** [comment]
> In the documentation for this method, you can see that there are two MIME types available as Response examples.

Actually I also can't see that. At least not on Apiary. I have to write in the header, and there is no statement explaining that there are different ones supported.

**Comment by balhoff** [comment]
Plain text is the default so that bare links to service results can be emailed to Phenoscape personnel, with the result that they can click the link and see a table of data. I expect that most people using JSON will be accessing it programmatically and so could specify the content-type without much trouble.

**Comment by hlapp** [comment]
> Plain text is the default so that bare links to service results can be emailed to Phenoscape personnel, with the result that they can click the link and see a table of data.

Then it should be the default for _all_ calls, and not for most calls except one.

**Comment by balhoff** [comment]
It is (or should be) the default for all calls where it is supported.

**Comment by hlapp** [comment]
> I expect that most people using JSON will be accessing it programmatically and so could specify the content-type without much trouble.

Most people using JSON will first inspect the documentation (isn't that what it's for?), and get a result they are not expecting.

**Comment by hlapp** [comment]
> It is (or should be) the default for all calls where it is supported.

This should not be conditional. If a method is called with a response type that's unsupported, it should return an error, not a different response type unless the Accept header allows it.

**Comment by balhoff** [comment]
Have you encountered an API method that does not return an error if you call it with an unsupported response type?

**Comment by balhoff** [comment]
@hlapp to see the response types supported by a given service, click the service name, then in the detail panel scroll down to the Response section. Example responses are listed by MIME type.

**Comment by hlapp** [comment]
> Have you encountered an API method that does not return an error if you call it with an unsupported response type?

Perhaps I should have phrased this better. I did encounter methods that behave inconsistently when called with no response type.

> to see the response types supported by a given service, click the service name, then in the detail panel scroll down to the Response section. Example responses are listed by MIME type.

I actually wasn't ever aware that there are more response types following after the first one, and that that fact would have a bearing on how I needed to operate the Apiary interface for testing example queries.

To go back to my principle points, _my comment and this issue are all about usability_. It shouldn't take submitting an issue and a whole ensuing thread to understand what's happening and how to use the API or understand the documentation. The API methods should behave consistently, and defaults should be the same from one call to another as much as possible. Not specifying a response type should result in the same consistent default, not one that changes depending on which response types are supported. What if you support only one type today and default to that and then add a second down the road and then change the default to the new one. Now you changed behavior and this will be difficult to debug.

Perhaps what I'm saying doesn't make sense. I reporting here from a users' perspective. I consider myself literate enough that the issues that I run into other users may run into as well. How can we save other potential users time and frustrations.
